components/hab/src/main.rs

```diff
@@ -287,7 +287,8 @@ fn sub_origin_key_upload(ui: &mut UI, m: &ArgMatches) -> Result<()> {
 
 fn sub_pkg_binlink(ui: &mut UI, m: &ArgMatches) -> Result<()> {
     let ident = PackageIdent::from_str(m.value_of("PKG_IDENT").unwrap())?;
-    let dest_dir = Path::new(m.value_of("DEST_DIR").unwrap_or(DEFAULT_BINLINK_DIR));
+    let env_or_default = henv::var(BINLINK_DIR_ENVVAR).unwrap_or(DEFAULT_BINLINK_DIR.to_string());
```

Comment:
Should you call the function you created above? `default_binlink_dir` which appears to be unused right now.